Unable to find angular.json when working directory is not the same as the project directory

### Describe the bug

When running `mvn clean install` inside the project directory, Quinoa is able to find the `angular.json` file without problem.
However, if I try to run `mvn clean install -f path/to/project/pom.xml`, Quinoa fails during the build phase with the following error:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.17.3:build (default) on project example-project: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkiverse.quinoa.deployment.QuinoaProcessor#processBuild threw an exception: java.lang.RuntimeException: Quinoa failed to read the angular.json file. %s
[ERROR]         at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.readAngularJson(AngularFramework.java:59)
[ERROR]         at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.lambda$buildDir$2(AngularFramework.java:33)
[ERROR]         at java.base/java.util.Optional.orElseGet(Optional.java:364)
[ERROR]         at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.buildDir(AngularFramework.java:32)
[ERROR]         at io.quarkiverse.quinoa.deployment.QuinoaProcessor.processBuild(QuinoaProcessor.java:184)
[ERROR]         at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:499)
[ERROR] Caused by: java.nio.file.NoSuchFileException: .\angular.json
[ERROR]         at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:85)
[ERROR]         at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
[ERROR]         at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
[ERROR]         at java.base/sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:234)
[ERROR]         at java.base/java.nio.file.Files.newByteChannel(Files.java:379)
[ERROR]         at java.base/java.nio.file.Files.newByteChannel(Files.java:431)
[ERROR]         at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:420)
[ERROR]         at java.base/java.nio.file.Files.newInputStream(Files.java:159)
[ERROR]         at io.quarkiverse.quinoa.deployment.framework.override.AngularFramework$1.readAngularJson(AngularFramework.java:56)
```

Our Angular application is located in the same directory as the maven module, hence the `.\angular.json` path. Our application.properties contains the following line: 
```
quarkus.quinoa.ui-dir=./
```

After quickly looking at the code, I would think something needs to be adjusted in the [AngularFramework.java](https://github.com/quarkiverse/quarkus-quinoa/blob/main/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java) file.

### Quinoa version

2.5.0

### Quarkus version

3.17.3

### Build / Runtime

Angular

### Package Manager

NPM

### Steps to reproduce the behavior

1. Set the `quarkus.quinoa.ui-dir` as described above.
2. Run `mvn clean install -f path/to/project/pom.xml`

### Expected behavior

The `angular.json` file should be found regardless of which directory is the working directory.

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java
@@ -106,7 +106,7 @@ public class QuinoaProcessor {
 
         initializeTargetDirectory(outputTarget);
 
-        final QuinoaConfig resolvedConfig = overrideConfig(launchMode, userConfig, packageJson);
+        final QuinoaConfig resolvedConfig = overrideConfig(launchMode, userConfig, packageJson, projectDirs.uiDir);
 
         return new ConfiguredQuinoaBuildItem(projectDirs.projectRootDir, projectDirs.uiDir, packageJson, resolvedConfig);
     }

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/FrameworkConfigOverrideFactory.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/FrameworkConfigOverrideFactory.java
@@ -1,5 +1,6 @@
 package io.quarkiverse.quinoa.deployment.framework;
 
+import java.nio.file.Path;
 import java.util.Optional;
 
 import jakarta.json.JsonObject;
@@ -13,5 +14,5 @@ public interface FrameworkConfigOverrideFactory {
     String getDefaultDevScriptName();
 
     QuinoaConfig override(QuinoaConfig delegate, Optional<JsonObject> packageJson, Optional<String> detectedDevScript,
-            boolean isCustomized);
+            boolean isCustomized, Path uiDir);
 }

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/FrameworkType.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/FrameworkType.java
@@ -72,9 +72,10 @@ public enum FrameworkType {
         return factory;
     }
 
-    public static QuinoaConfig overrideConfig(LaunchModeBuildItem launchMode, QuinoaConfig config, Path packageJsonFile) {
+    public static QuinoaConfig overrideConfig(LaunchModeBuildItem launchMode, QuinoaConfig config, Path packageJsonFile,
+            Path uiDir) {
         if (!config.framework().detection()) {
-            return UNKNOWN_FRAMEWORK.override(config, Optional.empty(), Optional.empty(), true);
+            return UNKNOWN_FRAMEWORK.override(config, Optional.empty(), Optional.empty(), true, uiDir);
         }
 
         final JsonObject packageJson = readPackageJson(packageJsonFile);
@@ -82,14 +83,14 @@ public enum FrameworkType {
 
         if (detectedFramework.isEmpty()) {
             LOG.trace("Quinoa could not auto-detect the frameworkType from package.json file.");
-            return UNKNOWN_FRAMEWORK.override(config, Optional.of(packageJson), Optional.empty(), true);
+            return UNKNOWN_FRAMEWORK.override(config, Optional.of(packageJson), Optional.empty(), true, uiDir);
         }
 
         final FrameworkType frameworkType = detectedFramework.get().type;
         LOG.infof("Quinoa detected '%s' frameworkType from package.json file.", frameworkType);
 
         return frameworkType.factory.override(config, Optional.of(packageJson), Optional.of(detectedFramework.get().devScript),
-                detectedFramework.get().isCustomized);
+                detectedFramework.get().isCustomized, uiDir);
     }
 
     private static JsonObject readPackageJson(Path packageJsonFile) {

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/AngularFramework.java
@@ -11,7 +11,6 @@ import io.quarkiverse.quinoa.deployment.config.PackageManagerCommandConfig;
 import io.quarkiverse.quinoa.deployment.config.QuinoaConfig;
 import io.quarkiverse.quinoa.deployment.config.delegate.PackageManagerCommandConfigDelegate;
 import io.quarkiverse.quinoa.deployment.config.delegate.QuinoaConfigDelegate;
-import io.quarkus.logging.Log;
 
 public class AngularFramework extends GenericFramework {
 
@@ -24,13 +23,14 @@ public class AngularFramework extends GenericFramework {
 
     @Override
     public QuinoaConfig override(QuinoaConfig originalConfig, Optional<JsonObject> packageJson,
-            Optional<String> detectedDevScript, boolean isCustomized) {
+            Optional<String> detectedDevScript, boolean isCustomized, Path uiDir) {
         final String devScript = detectedDevScript.orElse(getDefaultDevScriptName());
-        return new QuinoaConfigDelegate(super.override(originalConfig, packageJson, detectedDevScript, isCustomized)) {
+        return new QuinoaConfigDelegate(super.override(originalConfig, packageJson, detectedDevScript, isCustomized,
+                uiDir)) {
             @Override
             public Optional<String> buildDir() {
                 return Optional.of(originalConfig.buildDir().orElseGet(() -> {
-                    final JsonObject angularJson = readAngularJson(originalConfig);
+                    final JsonObject angularJson = readAngularJson(uiDir);
                     final JsonObject projectList = angularJson.getJsonObject("projects");
                     final JsonObject builder = projectList.values().stream()
                             .map(JsonValue::asJsonObject)
@@ -50,10 +50,8 @@ public class AngularFramework extends GenericFramework {
                 }));
             }
 
-            private static JsonObject readAngularJson(QuinoaConfig configuration) {
-                Log.debug("=== Configuration ===" + configuration);
-                try (JsonReader reader = Json
-                        .createReader(Files.newInputStream(Path.of(configuration.uiDir() + "/" + ANGULAR_JSON_FILE)))) {
+            private static JsonObject readAngularJson(Path uiDir) {
+                try (JsonReader reader = Json.createReader(Files.newInputStream(uiDir.resolve(ANGULAR_JSON_FILE)))) {
                     return reader.readObject();
                 } catch (IOException | JsonException e) {
                     throw new RuntimeException("Quinoa failed to read the angular.json file. %s", e);

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/GenericFramework.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/GenericFramework.java
@@ -3,6 +3,7 @@ package io.quarkiverse.quinoa.deployment.framework.override;
 import static io.quarkiverse.quinoa.deployment.config.PackageManagerCommandConfig.DEFAULT_DEV_SCRIPT_NAME;
 import static io.quarkiverse.quinoa.deployment.config.QuinoaConfig.DEFAULT_BUILD_DIR;
 
+import java.nio.file.Path;
 import java.util.Optional;
 
 import jakarta.json.JsonObject;
@@ -53,7 +54,7 @@ public class GenericFramework implements FrameworkConfigOverrideFactory {
 
     @Override
     public QuinoaConfig override(QuinoaConfig originalConfig, Optional<JsonObject> packageJson,
-            Optional<String> detectedDevScript, boolean isCustomized) {
+            Optional<String> detectedDevScript, boolean isCustomized, Path uiDir) {
         final String devScript = detectedDevScript.orElse(defaultScriptName);
         return new QuinoaConfigDelegate(originalConfig) {
             @Override

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/NextFramework.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/NextFramework.java
@@ -1,5 +1,6 @@
 package io.quarkiverse.quinoa.deployment.framework.override;
 
+import java.nio.file.Path;
 import java.util.Optional;
 
 import jakarta.json.JsonObject;
@@ -22,8 +23,7 @@ public class NextFramework extends GenericFramework {
 
     @Override
     public QuinoaConfig override(QuinoaConfig delegate, Optional<JsonObject> packageJson,
-            Optional<String> detectedDevScript,
-            boolean isCustomized) {
+            Optional<String> detectedDevScript, boolean isCustomized, Path uiDir) {
         LOG.warn("Next.js version 13 and above are not fully supported yet. Please make sure to use version 12 or below.");
 
         if (delegate.packageManagerCommand().build().orElse("???").equals("run build") && packageJson.isPresent()) {
@@ -43,7 +43,7 @@ public class NextFramework extends GenericFramework {
             }
         }
 
-        return new QuinoaConfigDelegate(super.override(delegate, packageJson, detectedDevScript, isCustomized)) {
+        return new QuinoaConfigDelegate(super.override(delegate, packageJson, detectedDevScript, isCustomized, uiDir)) {
 
             @Override
             public DevServerConfig devServer() {

--- a/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/ReactFramework.java
+++ b/deployment/src/main/java/io/quarkiverse/quinoa/deployment/framework/override/ReactFramework.java
@@ -1,5 +1,6 @@
 package io.quarkiverse.quinoa.deployment.framework.override;
 
+import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
@@ -19,8 +20,8 @@ public class ReactFramework extends GenericFramework {
 
     @Override
     public QuinoaConfig override(QuinoaConfig delegate, Optional<JsonObject> packageJson, Optional<String> detectedDevScript,
-            boolean isCustomized) {
-        return new QuinoaConfigDelegate(super.override(delegate, packageJson, detectedDevScript, isCustomized)) {
+            boolean isCustomized, Path uiDir) {
+        return new QuinoaConfigDelegate(super.override(delegate, packageJson, detectedDevScript, isCustomized, uiDir)) {
 
             @Override
             public PackageManagerCommandConfig packageManagerCommand() {

